### ISSUE TITLE
XDG Base Directory Specification is not being followed

### ISSUE THREAD

**Issue by Avamander** [ohmyzsh/ohmyzsh#9543]
**Describe the bug**

OMZ does not follow the [XDG-BD specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html). Zsh's own behaviour is a separate issue.

This was previously instantly closed here: https://github.com/ohmyzsh/ohmyzsh/issues/7759

**To Reproduce**
Steps to reproduce the behavior, for example:
1. Install OMZ
2. Oh My Zsh places all its stuff under `$HOME/.oh-my-zsh/`

**Expected behavior**
OMZ's folder should be under `$XDG_DATA_HOME/oh-my-zsh/` and some parts in `$XDG_CACHE_HOME/oh-my-zsh`

**Desktop (please complete the following information):**

 - OS / Distro: Ubuntu 20.04
 - Latest ohmyzsh update?: Yes
 - ZSH Version: 5.8
 - Terminal emulator: Konsole

**Comment by CantoroMC** [comment]
Zsh expected behaviour is to place the file in $ZDOTDIR or in $HOME, if the previous is not defined.
Additionally, they allow you to specify a $ZSH variable before launching the installation script.
ZSH="$XDG_DATA_HOME/oh-my-zsh" sh install.sh

I will not call this a bug

**Comment by Avamander** [comment]
@CantoroMC 
The defaults should follow the agreed and widespread specifications, not in the reverse. It is a bug.

**Comment by CantoroMC** [comment]
I also care about not have my home folder full of configuration file.
But as I explain before the only file not respectful of xdg  specification is .zshenv where you could specify ZDOTDIR.
The issue is not related to Omz but rather to the shell itself.

**Comment by Avamander** [comment]
@CantoroMC 

> But as I explain before the only file not respectful of xdg specification is .zshenv where you could specify ZDOTDIR.

Actually no, my current install has an `~/.oh-my-zsh` folder that is vital for the functioning of it. The defaults aren't correct.

**Comment by CantoroMC** [comment]
Ok, good luck

**Comment by Pound-Hash** [comment]
> Zsh expected behaviour is to place the file in $ZDOTDIR or in $HOME, if the previous is not defined. Additionally, they allow you to specify a $ZSH variable before launching the installation script. ZSH="$XDG_DATA_HOME/oh-my-zsh" sh install.sh
> 
> I will not call this a bug

I have the $ZDOTDIR variable defined to my .config/zsh folder. The oh-my-zsh script still installed everything into my home folder.

**Comment by carlosala** [comment]
You can pass the variable `$ZSH` to the installer to change where to install it. For example, in your case:
```sh
ZSH="~/.config/zsh/oh-my-zsh" sh installer.sh
```

**Comment by mikelpr** [comment]
@carlosala It should default to XDG_DATA_HOME/oh-my-zsh then and if it isn't defined fallback to $HOME/.oh-my-zsh, instead of always ~/.oh-my-zsh unless you define that envvar since most people just run the command that the green Install oh my zsh button shows at the top of the page, and anyways makes much more sense to use the XDG dirs if available instead of defaulting to polluting

**Comment by bavalpey** [comment]
> [@carlosala](https://github.com/carlosala) It should default to XDG_DATA_HOME/oh-my-zsh then and if it isn't defined fallback to $HOME/.oh-my-zsh

Nope, then you're still not following the spec. The spec says if XDG_DATA_HOME isn't defined

> $XDG_DATA_HOME defines the base directory relative to which user-specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

So it should be placed in $HOME/.local/share.

In fact, the *only* time it should be placed in $HOME/.oh-my-zsh is if the user has that as their $ZSH environment variable.

**Comment by robbyrussell** [comment]
Thanks for filing this, and sorry it took so long to give you a straight answer.

Having gone back through the code: everything Oh My Zsh writes is already relocatable, so this is a question about defaults rather than something we're unable to do. Setting these before install (or before the `source $ZSH/oh-my-zsh.sh` line in your `.zshrc`) moves all of it:

```zsh
export ZDOTDIR="$XDG_CONFIG_HOME/zsh"       # .zshrc itself; the installer honors this
export ZSH="$XDG_DATA_HOME/oh-my-zsh"       # the install, and $ZSH_CUSTOM follows it
export ZSH_CACHE_DIR="$XDG_CACHE_HOME/oh-my-zsh"
export ZSH_COMPDUMP="$ZSH_CACHE_DIR/zcompdump"
```

Changing the *defaults* is the part we're not going to do. `~/.oh-my-zsh` is baked into a huge number of existing installs, and into more or less every blog post and dotfiles repo out there. Moving it would need a migration that behaves correctly on all of them, and the risk of getting that wrong is a lot bigger than the tidiness it buys.

So I'm closing this rather than let it sit any longer. The narrower pieces are still open on their own — the cache dir in #8808 and the zcompdump location in #13406 — and those are much lower risk than moving the install root.

🤖 Co-authored with Claude Code
